packages/react-core/src/components/Nav/NavGroup.tsx

```diff
@@ -20,12 +20,21 @@ export const NavGroup: React.FunctionComponent<NavGroupProps> = ({
   className = '',
   id = getUniqueId(),
   ...props
-}: NavGroupProps) => (
-  <section className={css(styles.navSection, className)} aria-labelledby={id} {...props}>
-    <h2 className={css(styles.navSectionTitle)} id={id}>
-      {title}
-    </h2>
-    <ul className={css(styles.navList, className)}>{children}</ul>
-  </section>
-);
+}: NavGroupProps) => {
+  if (!title && !props['aria-label']) {
+    // eslint-disable-next-line no-console
+    console.warn("For accessibility reasons an aria-label should be specified on nav groups if a title isn't");
+  }
+
+  return (
+    <section className={css(styles.navSection, className)} aria-labelledby={id} {...props}>
```

Comment:
should `aria-labelledby` be conditionally rendered if `title` is set?

Reply:
Yes I do believe so, good catch!